src/web/WebSession.C

```diff
@@ -680,9 +680,16 @@ bool WebSession::start(WebResponse *response)
 {
   try {
     app_ = controller_->doCreateApplication(this).release();
-    if (!app_->internalPathValid_)
-      if (response->responseType() == WebResponse::ResponseType::Page)
-        response->setStatus(404);
+    if (app_) {
+      if (!app_->internalPathValid_) {
+        if (response->responseType() == WebResponse::ResponseType::Page) {
+          response->setStatus(404);
+        }
+      }
+    }
+    else {
+      throw WException("WebSession::start: doCreateApplication returned a nullptr");
```

Comment:
I think it would be better to refer to publicly documented `ApplicationCreator`, since `doCreateApplication` is an implementation detail and we can't expect users to be familiar with it.